packit_service_fedmsg/consumer.py

```diff
@@ -36,6 +37,23 @@
 }
 KOJI_TOPICS = {"org.fedoraproject.prod.buildsys.task.state.change"}
 
+PUSH_TOPIC = "org.fedoraproject.prod.git.receive"
+
+
+def specfile_changed(body) -> bool:
+    """
+    Does the commit contain specfile change?
+    :param body: message body
+    :return: bool
+    """
+    files = reduce(
+        lambda val, key: val.get(key) if val else None,
+        ["commit", "stats", "files"],
+        body,
+    )
+    file_names = list(files.keys()) if files else []
+    return any(".spec" in file_name for file_name in file_names)
```

Comment:
I'm wondering if the Python interpreter is smart enough to stop constructing the list once the first file_name with ".spec" is found :thinking: 

Also: maybe `file_name.endswith(".spec")` would be more correct.

Reply:
should be evaluated lazy because of the generator :thinking:

Reply:
Yeah, functionally works, but the execution time is a little but higher with the any construct, compared to using a for-loop an returning from there. [Code](https://gist.github.com/csomh/ae177bc9b23ce519bd59b1af6b7c9f5d). Results (in seconds per 1M calls):

```
for loop, spec-file file in the beginning of the list: 1.1026607289968524
any, spec-file file in the beginning of the list: 1.6310043529956602
for loop, spec-file file in the middle of the list: 9.095828573015751
any, spec-file file in the middle of the list: 11.731628537992947
```

But probably the above could be easily called "early optimisation". I was just curious :man_shrugging: :smile:

Reply:
what about generator instead of explicit list? :thinking: `.keys()` returns iterator, so we basically don't need to have list of the keys

better than list, but still...:
```
for loop, spec-file file in the beginning of the iterator: 0.15259436299675144
any, spec-file file in the beginning of the iterator: 0.36171924200607464
for loop, spec-file file in the middle of the iterator: 0.15011202699679416
any, spec-file file in the middle of the iterator: 0.3568390000000363
```
```py
files = (f"file{i}" if i != 10 else "file.spec" for i in range(300))
print("for loop, spec-file file in the beginning of the iterator:", timeit.timeit(lambda: has_spec_for(files)))
print("any, spec-file file in the beginning of the iterator:", timeit.timeit(lambda: has_spec_any(files)))

files = (f"file{i}" if i != 100 else "file.spec" for i in range(300))
print("for loop, spec-file file in the middle of the iterator:", timeit.timeit(lambda: has_spec_for(files)))
print("any, spec-file file in the middle of the iterator:", timeit.timeit(lambda: has_spec_any(files)))
```